packages/ts/react-crud/src/autogrid.tsx

```diff
@@ -170,16 +171,56 @@ function createDataProvider<TItem>(
   };
 }
 
+interface ColumnConfigurationOptions {
+  visibleColumns?: string[];
+  noHeaderFilters?: boolean;
+  customColumns?: JSX.Element[];
```

Comment:
Asking this, since it also can have some effects on DX when using an autocompletion in the IDE and the type JSX.Element give an illusion that they can pass in anything element, while the code is assuming that they are `GridColumn`s and will break if they aren't.

Reply:
Anyway, not related to this PR, we can improve that in another PR.

Reply:
I looked that up some time ago, and it's not possible currently with Typescript. When you use JSX such as `<span></span>` in Typescript it kind of forgets the specific component type that the element was created from and only checks that you are passing some JSX element.